x-pack/plugins/infra/public/pages/metrics/hosts/components/kpis/tile.tsx

```diff
@@ -101,10 +86,30 @@ export const Tile = ({ id, title, layers, style, toolTip, ...props }: KPIChartPr
     () =>
       getExtraActions({
         timeRange: afterLoadedState.dateRange,
-        query: searchCriteria.query,
+        query: shouldUseSearchCriteria ? afterLoadedState.query : undefined,
         filters,
       }),
-    [afterLoadedState.dateRange, filters, getExtraActions, searchCriteria.query]
+    [
+      afterLoadedState.dateRange,
+      afterLoadedState.query,
+      filters,
+      getExtraActions,
+      shouldUseSearchCriteria,
+    ]
+  );
+
+  const handleBrushEnd = useCallback(
```

Comment:
Nit: Maybe we can share this function between the 2 components?

Reply:
Definitely. These 2 components share many things. we need to try to centralize their implementation as much as possible. Best to do that in a separate ticket.